vars/piperPipelineStageInit.groovy

```diff
@@ -46,12 +46,21 @@ void call(Map parameters = [:]) {
     def script = checkScript(this, parameters) ?: this
     def utils = parameters.juStabUtils ?: new Utils()
     def stageName = StageNameProvider.instance.getStageName(script, parameters, this)
+    def scmInfo
+    
+    node(parameters.nodeLabel?:'master') {
+        deleteDir()
 
-    piperStageWrapper (script: script, stageName: stageName, stashContent: [], ordinal: 1, telemetryDisabled: true) {
-        def scmInfo = checkout scm
+        scmInfo = checkout scm
 
         setupCommonPipelineEnvironment script: script, customDefaults: parameters.customDefaults
 
+        stash allowEmpty: true, excludes: '', includes: '**', useDefaultExcludes: false, name: 'INIT'
+        script.commonPipelineEnvironment.configuration.stageStashes = [ (stageName): [ unstash : ["INIT"]]]
```

Comment:
I think the name of the stash is different depending on the pipeline, see for example [here](https://github.com/SAP/jenkins-library/blob/master/resources/com.sap.piper/pipeline/stashSettings.yml#L4). Not yet sure how this can be solved best.

Reply:
Here the stash is only used to be unstashed by the piperStageWrapper call below.
The stash config is resolved later (with the call in line 75) and based on the parameter `stashSettings`, which we could hand in as a parameter when calling the `piperPipelineStageInit` or via defaults in our configuration.

Reply:
Ok, makes sense. The INIT stash would hang around though, correct?

Reply:
Yes but I think that shouldn't be a problem, as this is the same in Cloud SDK Pipeline at the moment.

Reply:
I think the "INIT" stash is merged again after the Init stage is finished, and then re-used in other stages. While in the GP pipeline, a new stash "source" is created and used thereafter. Not sure if that is a problem worth considering or not.

Reply:
You are right, I remembered it wrongly. Sorry about the confusion.